terraform/staging/main.tf

```diff
@@ -26,4 +26,24 @@ module "networking" {
   source = "../modules/networking"
 
   default_tags = var.default_tags
-}
\ No newline at end of file
+}
+
+resource "aws_key_pair" "bastion_ssh_key" {
+  key_name   = "bastion_ssh_key"
+  public_key = file("delta_test.pub")
+}
+
+module "bastion" {
+  source = "git::https://github.com/Softwire/terraform-bastion-host-aws?ref=34554d5b603e97d3adb2e06bcdf3b02d9d2a7e95"
+
+  region                  = "eu-west-1"
+  name_prefix             = "stg"
+  vpc_id                  = module.networking.vpc.id
+  public_subnet_ids       = [for subnet in module.networking.public_subnets : subnet.id]
+  instance_subnet_ids     = [for subnet in module.networking.private_subnets : subnet.id]
+  admin_ssh_key_pair_name = aws_key_pair.bastion_ssh_key.key_name
+  external_allowed_cidrs  = ["31.221.86.178/32", "167.98.33.82/32", "82.163.115.98/32", "87.224.105.250/32", "87.224.18.46/32"]
```

Comment:
We should have this list as a variable because it's also used in modules/active_directory/security_groups for access to the AD management server. And in that case I missed out the Cambridge office IP.

Alternatively, slightly more secure would be to make the AD management box only accessible via this bastion box. In which case, can you add instructions for setting up SSH tunnelling for the RDP connection?

Reply:
Done 🙂